src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/HardwareIntrinsicHelpers.Aot.cs

```diff
@@ -49,12 +49,46 @@ public static MethodIL EmitIsSupportedIL(MethodDesc method, FieldDesc isSupporte
             var emit = new ILEmitter();
             ILCodeStream codeStream = emit.NewCodeStream();
 
-            codeStream.Emit(ILOpcode.ldsfld, emit.NewToken(isSupportedField));
-            codeStream.EmitLdc(flag);
-            codeStream.Emit(ILOpcode.and);
-            codeStream.EmitLdc(0);
-            codeStream.Emit(ILOpcode.cgt_un);
-            codeStream.Emit(ILOpcode.ret);
+            switch (instructionSet)
+            {
+                case InstructionSet.X64_AVX10v1_V512:
+                case InstructionSet.X64_AVX10v1_V512_X64:
+                case InstructionSet.X64_AVX512F_VL:
+                case InstructionSet.X64_AVX512F_VL_X64:
+                case InstructionSet.X64_AVX512BW_VL:
+                case InstructionSet.X64_AVX512BW_VL_X64:
+                case InstructionSet.X64_AVX512CD_VL:
+                case InstructionSet.X64_AVX512CD_VL_X64:
+                case InstructionSet.X64_AVX512DQ_VL:
+                case InstructionSet.X64_AVX512DQ_VL_X64:
+                case InstructionSet.X64_AVX512VBMI_VL:
+                case InstructionSet.X64_AVX512VBMI_VL_X64:
```

Comment:
```suggestion
            if (!uint.IsPow2((uint)flag))
            {
```
You should be able to just check whether the flag has one or more bits.